workflows/diagnostics/fv3net/diagnostics/prognostic_run/shell.py

```diff
@@ -0,0 +1,197 @@
+    x = state.data_2d
+    avg = vcm.weighted_average(x[variable], x.area, ["x", "y", "tile"])
+    avg.plot()
+    state.tape.save_plot()
+
+
+def set_iterm_tape(state: State):
+    state.tape = ItermTape()
+
+
+def hovmoller(state: State, variable, vmin=None, vmax=None):
+    z = state.data_2d[variable]
+    avg = vcm.zonal_average_approximate(state.data_2d.lat, z)
+    vmin = None if vmin is None else float(vmin)
+    vmax = None if vmax is None else float(vmax)
+    avg.plot(x="time", vmin=vmin, vmax=vmax)
+    state.tape.save_plot()
+
+
+commands = {"iterm": set_iterm_tape, "avg2d": avg2d, "hovmoller": hovmoller}
+
+
+class ProgShell(cmd.Cmd):
+    intro = (
+        "Welcome to the ProgRunDiag shell.   Type help or ? to list commands.\n"  # noqa
+    )
+
+    def do_avg2d(self, arg):
+        avg2d(loop_state, arg)
+
+    def do_iterm(self, arg):
+        set_iterm_tape(loop_state)
+
+    def do_hovmoller(self, arg):
+        hovmoller(loop_state, *arg.split())
+
+    def do_load(self, arg):
+        url = arg
+        loop_state.load(url)
+
+    def do_set(self, arg):
+        key, val = arg.split()
+        state[key] = val
+
+    def do_print(self, arg):
+        loop_state.print()
+
+    def do_meridional(self, arg):
+        variable = arg
+        time = int(state.get("time", "0"))
+        lon = int(state.get("lon", "0"))
+        transect = meridional_transect(
+            loop_state.data_3d.isel(time=time).merge(grid), lon
+        )
+        transect = transect.assign_coords(lon=lon)
+        transect[variable].plot(yincrease=False, y="pressure")
+        loop_state.tape.save_plot()
+
+    def do_zonal(self, arg):
+        variable = arg
+        time = int(state.get("time", "0"))
+        lat = float(state.get("lat", 0))
+        ds = loop_state.data_3d.isel(time=time).merge(grid)
+
+        transect_coords = vcm.select.zonal_ring(lat=lat)
+        transect = vcm.interpolate_unstructured(ds, transect_coords)
+        transect = transect.swap_dims({"sample": "lon"})
+        transect = transect.assign_coords(lat=lat)
+
+        plt.figure(figsize=(10, 3))
+        transect[variable].plot(yincrease=False, y="pressure")
+        loop_state.tape.save_plot()
+
+    def onecmd(self, line):
+        try:
+            super().onecmd(line)
+        except Exception as e:
+            print(e)
+
+    def do_map2d(self, arg):
+        variable = arg
+        time = int(state.get("time", "0"))
+        data = loop_state.data_2d.isel(time=time)
+        fv3viz.plot_cube(data, variable)
+        time_name = data.time.item().isoformat()
+        plt.title(f"{time_name} {variable}")
+        plt.tight_layout()
+        loop_state.tape.save_plot()
+
+    def do_exit(self, arg):
+        sys.exit(0)
+
+    def do_eval(self, arg):
+        f = pathlib.Path(arg)
+        for line in f.read_text().splitlines():
+            self.onecmd(line)
+
+
+def register_parser(subparsers):
+    parser = subparsers.add_parser("shell")
```

Comment:
Add a short help description here.

Reply:
Sorry, I'm not sure what you mean. Do you mean I should add help to the CLI or once the CLI opens?

Reply:
Yes. Does not need to be comprehensive, but a short usage description (in the form of the `help='...'` kwarg) would be helpful. Right now this command is not very visible when you do `prognostic_run_diags --help`:
```
usage: prognostic_run_diags [-h] [--log-level LOG_LEVEL] {save,log-viewer,metrics,movie,shell,piggy,report,report-from-urls,report-from-json} ...

Prognostic run diagnostics

positional arguments:
  {save,log-viewer,metrics,movie,shell,piggy,report,report-from-urls,report-from-json}
                        Prognostic run diagnostics
    save                Compute the prognostic run diags.
    log-viewer          Webapp for plotting time series information from the standard output of segmented FV3 runs.
    metrics             Compute metrics from verification diagnostics. Prints to standard output.
    movie               Generate movies from prognostic run.
    piggy               Log piggy backed metrics for prognostic run named TAGto weights and biases.
    report              Generate a static html report.
    report-from-urls    Generate a static html report from list of diagnostics.
    report-from-json    Generate report from diagnostics listed in given JSON file.

optional arguments:
  -h, --help            show this help message and exit
  --log-level LOG_LEVEL
```

Reply:
ok. for some reason the vscode github client was showing this comment at a different location a few 10s of lines higher up...which is why i was confused.